src/core/utils.ts

```diff
@@ -173,18 +173,18 @@ export function base64UrlEncode(buffer: ArrayBuffer): string {
     return urlVariant;
 }
 
-export function registerServiceWorker(
-    path: string
-): Promise<ServiceWorkerRegistration> {
+export async function registerServiceWorker(workerPath: string){
     if (!('serviceWorker' in navigator)) {
-        return Promise.reject(
-            'ServiceWorker is not supported in this browser, aborting...'
-        );
+        console.error('ServiceWorker is not supported in this browser, aborting...');
+        return;
     }
 
-    return navigator.serviceWorker.register(path).then(() => {
-        return navigator.serviceWorker.ready;
-    });
+    const updatedPath = new URL(workerPath, location.origin).href; 
```

Comment:
Any value in extracting the building of full SW URL to a util? (minor duplication in other places)

Reply:
Done